Sources/ManageServer.php

```diff
@@ -213,6 +227,141 @@ function ModifyGeneralSettings($return_config = false)
 }
 
 /**
+ * Align URLs with SSL Setting.
+ *
+ * If force_ssl has changed, ensure all URLs are aligned with the new setting.
+ * This includes:
+ *     - $boardurl
+ *     - $modSettings['smileys_url']
+ *     - $modSettings['avatar_url']
+ *     - $modSettings['custom_avatar_url'] - if found
+ *     - theme_url - all entries in the themes table
+ *     - images_url - all entries in the themes table
+ *
+ * This function will NOT overwrite URLs that are not subfolders of $boardurl.
+ * The admin must have pointed those somewhere else on purpose, so they must be updated manually.
+ * 
+ * A word of caution: You can't trust the http/https scheme reflected for these URLs in $globals
+ * (e.g., $boardurl) or in $modSettings.  This is because SMF may change them in memory to comply
+ * with the force_ssl setting - a soft redirect may be in effect...  Thus, conditional updates
+ * to these values do not work.  You gotta just brute force overwrite them based on force_ssl.
+ *
+ * @param int $new_force_ssl is the current force_ssl setting.
+ * @return void Returns nothing, just does its job
+ */
+function AlignURLsWithSSLSetting($new_force_ssl = 0)
+{
+	global $boardurl, $modSettings, $sourcedir, $db_prefix, $smcFunc;
+	require_once($sourcedir . '/Subs-Admin.php');
+
+	// Check $boardurl
+	if ($new_force_ssl == 2)
+		$newval = strtr($boardurl, array('http://' => 'https://'));
+	else
+		$newval = strtr($boardurl, array('https://' => 'http://'));
+	updateSettingsFile(array('boardurl' => '\'' . addslashes($newval) . '\''));
+
+	$new_settings = array();
+
+	// Check $smileys_url, but only if it points to a subfolder of $boardurl
+	if (BoardurlMatch($modSettings['smileys_url'])) {
```

Comment:
Nit: SMF's coding style requires a line break before the `{`

... I honestly don't know *why* we do it that way, but thus did the Ancient Powers declare that it should be done.

Reply:
Done.